f5_openstack_agent/lbaasv2/drivers/bigip/listener_service.py

```diff
@@ -299,28 +301,34 @@ def remove_session_persistence(self, service, bigips):
         and load balancer definition.
         :param bigips: Single BigIP instances to update.
         """
-        pool = service["pool"]
-        if "session_persistence" in pool and pool['session_persistence']:
-            vip = self.service_adapter.get_virtual_name(service)
-            vip["persist"] = []
-            vip["fallbackPersistence"] = ""
-            persistence = pool['session_persistence']
-            persistence_type = persistence['type']
-            listener = service["listener"]
-            if listener['protocol'] == 'TCP':
-                # Revert VS back to fastL4. Must do an update to replace
-                # profiles instead of using add/remove profile. Leave http
-                # profiles in place for non-TCP listeners.
-                vip['profiles'] = ['/Common/fastL4']
 
-            for bigip in bigips:
-                # remove persistence (and revert profiles if TCP)
+        vip = self.service_adapter.get_virtual_name(service)
+        vip["persist"] = []
+        vip["fallbackPersistence"] = ""
+
+        listener = service["listener"]
+        if listener['protocol'] == 'TCP':
+            # Revert VS back to fastL4. Must do an update to replace
+            # profiles instead of using add/remove profile. Leave http
+            # profiles in place for non-TCP listeners.
+            vip['profiles'] = ['/Common/fastL4']
+
+        for bigip in bigips:
+            # Check for custom app_cookie profile.
+            has_app_cookie = False
+            vs = self.vs_helper.load(bigip, vip['name'], vip['partition'])
+            persistence = getattr(vs, 'persist', None)
+            if persistence:
+                persist_name = persistence[0].get('name', '')
+                if persist_name:
+                    has_app_cookie = persist_name.startswith('app_cookie')
```

Comment:
The old implementation looked for 'APP_COOKIE' in the type.  Is the persist_name guaranteed to be lower-case?  Should .lower() be in the call chain?